### ISSUE TITLE
`The drag and drop` function is no longer functional in the latest code

### ISSUE THREAD

**Issue by wszgrcy** [microsoft/vscode#219047]
Does this issue occur when all extensions are disabled?: Yes

work success
```
Version: 1.90.2
Commit: Unknown
Date: Unknown
Electron: 29.4.0
ElectronBuildId: undefined
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 6.5.0-41-generic
```
work failed
```
Version: 1.91.0
Commit: Unknown
Date: Unknown
Electron: 29.4.0
ElectronBuildId: undefined
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 6.5.0-41-generic
```
This issue occurs when opening a webview type window while developing extensions
Steps to Reproduce:

1. I use mdn demo https://mdn.github.io/dom-examples/drag-and-drop/copy-move-DataTransfer.html
2. add html to webview html
3. open webview .`dragover` loss,and can't drop

I use `tag 1.90.2` compile and run vscode-oss work .but use latest commit `aea213b7fcc7de5c24ad797ac1af209b159d451f` doesn't work.

**Comment by gjsjohnmurray** [comment]
Does the technique described at https://github.com/microsoft/vscode/issues/213045#issuecomment-2158437545 get you to the exact commit where the problem appeared?

**Comment by wszgrcy** [comment]
> Does the technique described at [#213045 (comment)](https://github.com/microsoft/vscode/issues/213045#issuecomment-2158437545) get you to the exact commit where the problem appeared?

https://github.com/microsoft/vscode/commit/51917e869e4d643f9ecd5cfc6b618ba264d905c1
After constantly switching commit. I found that after submitting '51917e869e4d643f9ecd5cfc6b618ba264d905c1', there was an abnormal drag in the webview
And according to the submitted information, it was found that when dragging inside the 'webview', it is necessary to hold down the 'shift' to drop it inside the 'webview'. Is this a normal operating behavior
Cannot hold down 'shift' before dragging, it is necessary to hold down while dragging, which is a strange operation behavior

**Comment by swordensen** [comment]
> And according to the submitted information, it was found that when dragging inside the 'webview', it is necessary to hold down the 'shift' to drop it inside the 'webview'. Is this a normal operating behavior

I believe it is expected operating behavior

@mjbvz said this on the original issue

> Here's the behavior we want:

If the user drags a file over a webview without holding shift, they see the editor drop overlay. Dropping the file opens in that editor group
Once the user presses shift while dragging, they instead start dropping the content into the webview. The drop overlay should now go away
Once the user releases shift, the drop overlay shows again

Also the tooltip indicates that the user should _hold_ shift to drop a file into the editor 

![image](https://github.com/microsoft/vscode/assets/21202515/45b94b5a-268d-4c2e-8551-84a151b3844c)

**Comment by ZoolMons** [comment]
> > And according to the submitted information, it was found that when dragging inside the 'webview', it is necessary to hold down the 'shift' to drop it inside the 'webview'. Is this a normal operating behavior
> 
> I believe it is expected operating behavior
> 
> @mjbvz said this on the original issue #182449
> 
> > Here's the behavior we want:
> 
> > If the user drags a file over a webview without holding shift, they see the editor drop overlay. Dropping the file opens in that editor group
> > Once the user presses shift while dragging, they instead start dropping the content into the webview. The drop overlay should now go away
> > Once the user releases shift, the drop overlay shows again
> 
> Also the tooltip indicates that the user should _hold_ shift to drop a file when using the editor view. Maybe we should include a similar tooltip when dropping into webviews?
> 
> ![image](https://private-user-images.githubusercontent.com/21202515/345153830-45b94b5a-268d-4c2e-8551-84a151b3844c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5OTA4OTksIm5iZiI6MTcxOTk5MDU5OSwicGF0aCI6Ii8yMTIwMjUxNS8zNDUxNTM4MzAtNDViOTRiNWEtMjY4ZC00YzJlLTg1NTEtODRhMTUxYjM4NDRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDA3MDk1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlhYmFjYTA0MGRiMjQ4Y2ZmYzFhNGNjM2IzNTUxOTI1NTUxOGUxOWIzM2I0MWY2NzlkOTgwN2U5ZmY1YmQ4OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KlVMvwtcAB2trDkOlPLMkZsWEzSuz69MRn39efWGLEs)

Hello swordensen,

I also meet this situation when I want to drag files on to webView, and I saw you already fixed and merged in https://github.com/microsoft/vscode/pull/209211. But I didn't find related information in the change log from v1.90.1 and v1.90.2. Do you have a plan to release the pull request?

**Comment by wszgrcy** [comment]
> > And according to the submitted information, it was found that when dragging inside the 'webview', it is necessary to hold down the 'shift' to drop it inside the 'webview'. Is this a normal operating behavior
> 
> I believe it is expected operating behavior
> 
> @mjbvz said this on the original issue #182449
> 
> > Here's the behavior we want:
> 
> > If the user drags a file over a webview without holding shift, they see the editor drop overlay. Dropping the file opens in that editor group
> > Once the user presses shift while dragging, they instead start dropping the content into the webview. The drop overlay should now go away
> > Once the user releases shift, the drop overlay shows again
> 
> Also the tooltip indicates that the user should _hold_ shift to drop a file when using the editor view. Maybe we should include a similar tooltip when dropping into webviews?
> 
> ![image](https://private-user-images.githubusercontent.com/21202515/345153830-45b94b5a-268d-4c2e-8551-84a151b3844c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAwMDM3NzYsIm5iZiI6MTcyMDAwMzQ3NiwicGF0aCI6Ii8yMTIwMjUxNS8zNDUxNTM4MzAtNDViOTRiNWEtMjY4ZC00YzJlLTg1NTEtODRhMTUxYjM4NDRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDEwNDQzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1ZTcwMzUyM2EyMzM4YjQ3NjE2OTBlMDgxMjdlNzkzOTg1NzI5ZjMwZTA5YTIyYjRlZTRjZjIyMGUxNjcwNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bdr8INYLfOIeVUb9zqeO92Su_dckBJM-L9Kc0BOVvDM)

not editor=>webview or webview => editor;
Internal drag in `webview` will fail

**Comment by mjbvz** [comment]
Holding shift to drop is expected when dropping external content into a webview, such as from the explorer. However it's unexpected here as there's no file content on the drag event that could be opened in an editor. That's why you don't see even any overlay that mentions holding shift

I'm going to revert #209211 for now until we can get a more reliable fix for the original issue

**Comment by swordensen** [comment]
I see I didnt understand the issue. I didn't run into it myself because I was only dealing with dragging and dropping files. 

Maybe we should only bubble up the drag event if the user is dragging a file.

**Comment by swordensen** [comment]
Okay, so I updated my new PR so that only drag events coming from outside the editor with a file will be consumed by the new drag event listener. 

Now internal drag events will not be blocked. 


https://github.com/microsoft/vscode/assets/21202515/89032354-23c4-41a1-b9ab-09aa529884ce

**Comment by rzhao271** [comment]
I'm unable to view the full mp4s or download them on MS Edge on Windows. I wonder if they got corrupted.
The MDN demo seems to work on my end, now.
Verified.

**Comment by ssibrahimpur** [comment]
Kindly make an immediate release with the solution as this is blocking our production release for a plugin which uses drag drop feature.

**Comment by Yoyokrazy** [comment]
The issue has been closed and verified, so this issue should be fixed in the incoming Stable Recovery release, coming this week.

**Comment by ssibrahimpur** [comment]
[like]  Sachin Shivagonda Ibrahimpur reacted to your message:
________________________________

**Comment by daulis** [comment]
@mjbvz This problem was reintroduced in vscode 1.92.0

The way that I can reproduce is to do an interactive rebase (GUI) via GitLens

**Comment by swordensen** [comment]
I thought this might happen, if the maintainers want to merge in the recovery branch into main I don't mind handling conflicts on my fix PR https://github.com/microsoft/vscode/pull/219148

**Comment by stefanfrings2** [comment]
I also noticed that the problem was re-introduced in version 1.92.0. It worked fine in 1.91.0.
Related ticket in ThunderClient: #1567

**Comment by swordensen** [comment]
> I also noticed that the problem was re-introduced in version 1.92.0 using ThunderClient. It worked fine in vscode 1.91.0. Related issue in ThunderClient: [#1567](https://github.com/thunderclient/thunder-client-support/issues/1567)
> 
> Please re-open this issue.

afaik the bug was introduced in 1.91.0 resolved in 1.91.1 and re-introduced in 1.92.0 . A fix has been merged into main but, it has not been released. 

However, if you can re-create the issue by building from source with the latest changes please let me know and I will look into it. 

In the meantime, we have to wait for the next release/patch before the fix is available.

**Comment by ccastrotrejo** [comment]
Hi all, Logic Apps vscode extension is also having trouble with the drag and drop experience no working inside our own webview. This really hits out main experience in the extension, therefore users can't use it correctly. Could you please make a hotfix for the 1.92.x version?  . https://github.com/Azure/LogicAppsUX/issues/5265

**Comment by ccastrotrejo** [comment]
This is now fixed in vscode 1.92.1. Thank you so much!
